### DISCUSSION TITLE
Source Generator for reactive properties

### DISCUSSION THREAD

**Discussion by benjamin-hejl** [reactiveui/ReactiveUI#3674] [Ideas]
ReactiveUI is missing a modern property boiler plate code generator. Roslyn source code generator provides better usage (source code is visible, debuggable etc) with generating source code, than the ReactiveUI.Fody, which edits IL (afaik).

My idea is to generate boiler plate code like this:

```
[ReactiveProperty]
private int _intField;

// generates
public int IntField
{
    get => return _intField;
    set => this.RaiseAndSetIfChanged(ref _intField, value);
}

[ReactiveProperty]
private readonly ObservableAsPropertyHelper<Int> _intValue;

// generates
public int IntValue
{
    get => return _intValue.Value;
}
```
The `[ReactiveProperty]` attribute could provide some more hints to the generator - for example property name, whether both setter & getter should be generated (or their visibility).

**Comment by ChrisPulman** [comment]
Please use ReactiveUI.SourceGenerators with [Reactive], [ObservableAsProperty], [ReactiveCommand], [IViewFor] attributes to solve this.
